pyupgrade/_plugins/typing_pep563.py

```diff
@@ -0,0 +1,107 @@
+import ast
+import functools
+from collections import deque
+from typing import Iterable
+from typing import List
+from typing import MutableSequence
+from typing import NamedTuple
+from typing import Optional
+from typing import Sequence
+from typing import Tuple
+
+from tokenize_rt import Offset
+
+from pyupgrade._ast_helpers import ast_to_offset
+from pyupgrade._data import register
+from pyupgrade._data import State
+from pyupgrade._data import Token
+from pyupgrade._data import TokenFunc
+
+
+class Replacement(NamedTuple):
+    node: ast.expr
+    new: str
+
+
+def _supported_version(state: State) -> bool:
+    return (
+        state.settings.min_version >= (3, 10) or (
+            'annotations' in state.from_imports['__future__']
+        )
+    )
+
+
+def _dequote(i: int, tokens: List[Token], *, new: str) -> None:
+    tokens[i] = tokens[i]._replace(src=new)
+
+
+def _replace_string_literal(
+    annotation: Optional[ast.expr],
+) -> Sequence[Replacement]:
+    nodes: MutableSequence[ast.AST] = deque()
+    if annotation is not None:
+        nodes.append(annotation)
+    replacements: MutableSequence[Replacement] = deque()
+
+    while nodes:
+        node = nodes.pop()
+        if isinstance(node, ast.Subscript):
+            if isinstance(node.value, ast.Name):
+                if node.value.id not in {
+                    'Literal',
+                    'NamedTuple',
+                    'TypedDict',
+                    'Annotated',
+                }:
+                    nodes.append(node.slice)
+        elif isinstance(node, ast.Constant):  # pragma: no cover (py37)
+            if isinstance(node.value, str):
+                replacements.append(Replacement(node, node.value))
+        elif isinstance(node, ast.Str):  # pragma: no cover (py38+)
+            replacements.append(Replacement(node, node.s))
+        else:
+            for name in reversed(node._fields):
+                value = getattr(node, name)
+                if isinstance(value, ast.AST):
+                    nodes.append(value)
+                elif isinstance(value, list):
+                    nodes.extend(value)
+
+    return replacements
+
+
+@register(ast.FunctionDef)
+def visit_FunctionDef(
+    state: State,
+    node: ast.FunctionDef,
+    parent: ast.AST,
+) -> Iterable[Tuple[Offset, TokenFunc]]:
+    if not _supported_version(state):
+        return
+    replacements: MutableSequence[Replacement] = deque()
+    for args_name in node.args._fields:
+        args = getattr(node.args, args_name)
+        if args is None:
+            continue
+        for arg in args:
+            if hasattr(arg, 'annotation'):
+                value = getattr(arg, 'annotation')
+                replacements.extend(_replace_string_literal(value))
```

Comment:
this is very weird, the args always have `.annotation` so you shouldn't need to `getattr` here